crates/ironrdp-cliprdr/src/backend/windows/clipboard_data.rs

```diff
@@ -0,0 +1,59 @@
+use crate::backend::windows::{WinCliprdrError, WinCliprdrResult};
+use crate::pdu::ClipboardFormatId;
+
+use winapi::shared::{minwindef::HGLOBAL, winerror::ERROR_SUCCESS};
+use winapi::um::errhandlingapi::GetLastError;
+use winapi::um::winbase::{GlobalAlloc, GlobalFree, GlobalLock, GlobalUnlock, GMEM_MOVEABLE};
+use winapi::um::winuser::SetClipboardData;
+
+/// Safe wrapper around windows global clipboard data handle.
+pub struct ClipboardData(HGLOBAL);
+
+impl ClipboardData {
+    /// Renders data into the clipboard. Should be only invoked in the context of processing message
+    /// `WM_RENDERFORMAT` or `WM_RENDERALLFORMATS`.
+    pub fn render(format: ClipboardFormatId, data: &[u8]) -> WinCliprdrResult<()> {
```

Comment:
I would say that "render" is a good verb here, `SetClipboardData` can either promise data to the clipboard (by passing `NULL` for the `hMem` parameter) or actually place data into the clipboard. Since this is called in response to a `WM_RENDER*` message, it seems fair to name it `render`.

Reply:
Makes sense, thank you!